### DISCUSSION TITLE
elem.getAttribute('innerText') returns null in safari

### DISCUSSION THREAD

**Discussion by aparameswaran** [webdriverio/webdriverio#6994] [Q&A]
I am trying to get text by using the following line elem.getAttribute('innerText'). This works perfectly in chrome and in safari browser it returns null. Found some forums which suggested work around like textContent. Looking for help to see what is the equivalent workaround which works in safari browser.

**Comment by christian-bromann** [comment]
Maybe try:

```js
const elem = await $('#elem')
const text = browser.execute((elem) => elem.innerText, elem)
```

**Reply by aparameswaran** [reply]
This didn't help. I had to change it to elem.getText()
